geoh5py/workspace/workspace.py

```diff
@@ -810,32 +853,35 @@ def root(self) -> Optional[Entity]:
         """
         return self._root
 
-    @root.setter
-    def root(self, entity) -> Optional[Entity]:
-        if self._root is None:
-            assert isinstance(
-                entity, RootGroup
-            ), f"The given root entity must be of type {RootGroup}"
-            self._root = entity
-
-        return self._root
-
-    @staticmethod
-    def save_entity(entity: Entity, add_children: bool = True):
+    def save_entity(
+        self,
+        entity: Entity,
+        add_children: bool = True,
+        file: Optional[Union[str, h5py.File]] = None,
+    ):
         """
         Save or update an entity to geoh5.
 
         :param entity: Entity to be written to geoh5.
-        :param close_file: Close the geoh5 database after writing is completed.
         :param add_children: Add children entities to geoh5.
+        :param file: :obj:`h5py.File` or name of the target geoh5
         """
-        H5Writer.save_entity(entity, add_children=add_children)
+        self._io_call(file, H5Writer.save_entity, entity, add_children=add_children)
 
     @property
     def types(self) -> List["EntityType"]:
         """Get all active entity types registered in the workspace."""
         return self._all_types()
 
+    def validate_h5file(self, file) -> h5py.File:
+        """
+        Validate the h5file name
+        """
+        if file is None:
+            file = self.h5file
+
+        return fetch_h5_handle(file)
+
```

Comment:
this function is not necessary and prevents usage of context manager.
Can be replaced everywhere by `fetch_h5_handle(file or self.h5file)`

With `@contextmanager` on `fetch_h5_handle`, it becomes everywhere:
`with fetch_h5_handle(file or self.h5file) as h5file:`